transports/janus_mqtt.c

```diff
@@ -0,0 +1,648 @@
+void janus_mqtt_client_publish_janus_failure(void *context, MQTTAsync_failureData *response);
+int janus_mqtt_client_publish_admin_message(janus_mqtt_context *ctx, char *payload);
+void janus_mqtt_client_publish_admin_success(void *context, MQTTAsync_successData *response);
+void janus_mqtt_client_publish_admin_failure(void *context, MQTTAsync_failureData *response);
+void janus_mqtt_client_destroy_context(janus_mqtt_context *ctx);
+
+/* We only handle a single client */
+static janus_mqtt_context *context_ = NULL;
+
+int janus_mqtt_init(janus_transport_callbacks *callback, const char *config_path) {
+	if(callback == NULL || config_path == NULL) {
+		/* Invalid arguments */
+		return -1;
+	}
+
+	/* Initializing context */
+	janus_mqtt_context *ctx = g_malloc0(sizeof(struct janus_mqtt_context));
+	ctx->gateway = callback;
+	context_ = ctx;
+
+	/* Read configuration */
+	char filename[255];
+	g_snprintf(filename, 255, "%s/%s.cfg", config_path, JANUS_MQTT_PACKAGE);
+	JANUS_LOG(LOG_VERB, "Configuration file: %s\n", filename);
+	janus_config *config = janus_config_parse(filename);
+	if(config != NULL)
+		janus_config_print(config);
+
+	/* Handle configuration */
+	janus_config_item *url_item = janus_config_get_item_drilldown(config, "general", "url");
+	const char *url = g_strdup((url_item && url_item->value) ? url_item->value : "tcp://localhost:1883");
+
+	janus_config_item *client_id_item = janus_config_get_item_drilldown(config, "general", "client_id");
+	const char *client_id = g_strdup((client_id_item && client_id_item->value) ? client_id_item->value : "guest");
+
+	janus_config_item *username_item = janus_config_get_item_drilldown(config, "general", "username");
+	const char *username = g_strdup((username_item && username_item->value) ? username_item->value : "guest");
+	
+	janus_config_item *password_item = janus_config_get_item_drilldown(config, "general", "password");
+	const char *password = g_strdup((password_item && password_item->value) ? password_item->value : "guest");
+
+	janus_config_item *json_item = janus_config_get_item_drilldown(config, "general", "json");
+	if(json_item && json_item->value) {
+		/* Check how we need to format/serialize the JSON output */
+		if(!strcasecmp(json_item->value, "indented")) {
+			/* Default: indented, we use three spaces for that */
+			json_format_ = JSON_INDENT(3) | JSON_PRESERVE_ORDER;
+		} else if(!strcasecmp(json_item->value, "plain")) {
+			/* Not indented and no new lines, but still readable */
+			json_format_ = JSON_INDENT(0) | JSON_PRESERVE_ORDER;
+		} else if(!strcasecmp(json_item->value, "compact")) {
+			/* Compact, so no spaces between separators */
+			json_format_ = JSON_COMPACT | JSON_PRESERVE_ORDER;
+		} else {
+			JANUS_LOG(LOG_WARN, "Unsupported JSON format option '%s', using default (indented)\n", json_item->value);
+			json_format_ = JSON_INDENT(3) | JSON_PRESERVE_ORDER;
+		}
+	}
+
+	{ // connect configuration
```

Comment:
Use `/* ... */` instead of `// ...` for comments please (same for other occurrences).